diff -ru a/include/swift/AST/Stmt.h b/include/swift/AST/Stmt.h
--- a/include/swift/AST/Stmt.h	2022-09-21 12:56:54.000000000 +0200
+++ b/include/swift/AST/Stmt.h	2022-11-04 14:39:18.407971007 +0100
@@ -920,7 +920,6 @@
   /// Get the CaseStmt block to which the fallthrough transfers control.
   /// Set during Sema.
   CaseStmt *getFallthroughDest() const {
-    assert(FallthroughDest && "fallthrough dest is not set until Sema");
     return FallthroughDest;
   }
   void setFallthroughDest(CaseStmt *C) {
